### ISSUE TITLE
AssemblyConfigurationTest uses deprecated ObjectMapper.configure method

### ISSUE THREAD

**Issue by manusa** [eclipse/jkube#2526]
### Description

The class [AssemblyConfigurationTest](https://github.com/eclipse/jkube/blob/5da6068d7756784aef9569568ee80f98da97e296/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/AssemblyConfigurationTest.java) uses a deprecated method.

The following lines should be changed to (updating imports if necessary):

https://github.com/eclipse/jkube/blob/5da6068d7756784aef9569568ee80f98da97e296/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/AssemblyConfigurationTest.java#L263-L264

- [x] should be changed to:
```java
    final ObjectMapper mapper = JsonMapper.builder().configure(MapperFeature.USE_ANNOTATIONS, false).build();
```

### Before you start :red_circle:

:point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down::point_down::point_down::point_down::point_down::point_down::point_down:
Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first. Pay special attention to the ECA agreement section and the requirement to sign-off your commit.

#### Kubernetes

If you don't have a real Kubernetes cluster available (most probably), you can use Minikube or Kind to test with a local cluster.

- Minikube [get started guide](https://minikube.sigs.k8s.io/docs/start/)
- Kind (Requires Docker) [quick start](https://kind.sigs.k8s.io/docs/user/quick-start/)

#### OpenShift

If you don't have a real OpenShift cluster available (most probably), you can use [Red Hat's developer Sandbox for Red Hat OpenShift](https://developers.redhat.com/developer-sandbox). The only requirement is to have a Red Hat account.

Once you have your Sandbox environment, you'll need to download the oc tool from the cluster console.
(Press the `?` icon and from the context menu select `Command line tools`, you'll be redirected to https://$subdomain.openshiftapps.com/command-lines-tools where you'll be able to download the CLI for your platform)

**Comment by rohanKanojia** [comment]
> Do you have any blog about how to run and test my chances in localhost...?

@Yogeshknaik : Sorry, I don't understand. Could you please elaborate?

**Comment by Yogeshknaik** [comment]
@rohanKanojia  I followed  [Your blog](https://rohaan.medium.com/how-to-contribute-to-eclipse-jkube-f3bc7f902bf4)   but  I'm facing errors after  "mvn clean install".

**Comment by rohanKanojia** [comment]
@Yogeshknaik : Are you on windows? Our project does not compile on windows #1338

**Comment by ajayl83** [comment]
@rohanKanojia is this still open? if yes would like to work on this

**Comment by rohanKanojia** [comment]
@ajayl83 : yes, let me assign it to you.

**Comment by ajayl83** [comment]
@rohanKanojia i did fixed issue, but unfortunately some of the tests are failing..  not sure if i am missing any configurations ExternalCommandTest.execute_whenCommandFailed_thenThrowException:59 
Expecting message to be:
  "Process 'ls idontexist' exited with status 2"
but was:
  "Process 'ls idontexist' exited with status 1"

Throwable that failed the check:

java.io.IOException: Process 'ls idontexist' exited with status 1
	at org.eclipse.jkube.kit.common.ExternalCommand.execute(ExternalCommand.java:76)
	at org.eclipse.jkube.kit.common.ExternalCommand.execute(ExternalCommand.java:55)
	at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:63)
	at org.assertj.core.api.ThrowableTypeAssert.isThrownBy(ThrowableTypeAssert.java:59)
	at org.eclipse.jkube.kit.common.ExternalCommandTest.execute_whenCommandFailed_thenThrowException(ExternalCommandTest.java:58)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:728)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(Timeou

**Comment by rohanKanojia** [comment]
@ajayl83 : Are you using windows?

**Comment by rohanKanojia** [comment]
You're right, the test was disabled on windows. 

For this issue, it's okay to verify project complies cleanly with `-DskipTests`. 

Is it possible for you to create a new issue for this problem related to ExternalCommand on MacOS?

**Comment by ajayl83** [comment]
@rohanKanojia yes i will do that, it seems it also depends on processor? i am using M1

**Comment by rohanKanojia** [comment]
@ajayl83 : oh, I see. Our [Eclipse Jenkins CI instance](https://ci.eclipse.org/jkube/job/Build/) only verifies build on linux amd64 .

**Comment by ajayl83** [comment]
@rohanKanojia ok i opened the PR

**Comment by rohanKanojia** [comment]
@ajayl83 : Thanks, your changes look good. You need to complete some legal requirements before we can merge your pull request. 

> Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first. Pay special attention to the ECA agreement section and the requirement to sign-off your commit.
